### PR TITLE
Gourmet Command

### LINKED ISSUE #88: Connossieur Role and Command
As a member of the CS Dev Team, I want a role that gains access to a /where2eat (subject to change) command so that the Dev Team can decide where to eat lunch easily.

Features:
- [ ] Add Restaurant
- [ ] Remove Restaurant
- [ ] Randomly Select Restaurant
- [ ] Restaurants should go on a week long cooldown once selected
- [ ] Random Pool limited by type of food(?)

### PR DESCRIPTION
**Pull request by ReginPotter**
# Description

Added a command that will recommend restaurants to the user. The user can also add and remove restaurants from the list. The choices are currently vegan and normal.

## Issues

Closes #88

## Type of change

Select one or more of the following:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (describe below)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. List any edge cases you tested.
If you come up with any edge cases you didn't test while writing this PR, cancel the PR and test again.

- [x] Test A
       - `-feedme`
       - try out any button
- [x] Test B
       - `-feedme`
       - try out meat until there are no more restaurants
       - `-feedme`
       - try out meat once more

# Checklist:

- [X] All local commits have been pushed to remote
- [X] All changes on the base branch have been merged into this branch, either by rebase or merge
- [X] My code is [PEP-8](https://pep8.org/) compliant (excluding maximum line length, keep that to 100ish characters)
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added [Google Docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) for all new functions/methods
- [X] I have made corresponding changes to the documentation

### CODE REVIEW

File: Cogs/Gourmet.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        async def vegan(self,interaction:discord.Interaction,button:discord.ui.Button):
+            if not len(self.cog.vegan_restaurant) == 0:
+                await interaction.response.send_message(self.cog.vegan_restaurant.pop())
+            else:
+                await interaction.response.send_message('You have run out of restaurants. You are going to arbys.')
+
+    @commands.command()
+    @commands.has_permissions(administrator=True)
+    async def feedme(self, ctx):
+        self.normal_restaurant = []
```
Comment (mbozada) [inline Cogs/Gourmet.py:92]: I think a cleaner implementation would be to find a way to initialize these lists when the Cog is loaded.

Talk with Anmol about the work he's doing with faqbot. We read the file in the setup() function, and then pass some additional parameters to the Cog Initialization.

Once you have the file read in, feedme can work by making copies of the attributes:
`restaurants = self.normal_restaurants`

**Review by mbozada — CHANGES_REQUESTED** [review]
This is a great first step for the gourmet command. Good job Regin!

The big thing I want you to work on is reading in the file during setup and working off of a copy for the list.

A small thing that would be nice is if you could change the way the restaurants are posted. Right now the new messages with recommendations eventually push the button off the screen.
image

Maybe view has a line for text output? Every time a button is pushed the line gets edited with relevant output.

**Review by mbozada — APPROVED** [review]
This is a good start!

If you could go ahead and update issue #88 or create a new issue with our desired future features, then I think you're done for now!

**Comment by L30N1DAS** [comment]
Good work! The function does it's job very well. It doesn't seem added restaurants persist upon shutdown of the bot though and more functionality could definitely be added later. Can be pulled into master now however.
